test/test.go

```diff
@@ -110,13 +110,17 @@ func (req *Request) Validate() error {
 
 	req.Headers = mergeHeaders(make(http.Header), req.Headers)
 
-	if req.Method == "GET" {
+	switch req.Method {
+	case "GET":
 		if req.Body != "" {
 			return fmt.Errorf("GET requests must not have body, found \"%v\"", req.Body)
 		}
 		return nil
+	case "POST":
+		return nil
+	default:
```

Comment:
Is there a reason to only whitelist certain methods?

Reply:
Looks like unfinished code.